addons/stock/models/stock_inventory.py

```diff
@@ -231,6 +234,8 @@ def _get_inventory_lines_values(self):
             for void_field in [item[0] for item in product_data.items() if item[1] is None]:
                 product_data[void_field] = False
             product_data['theoretical_qty'] = product_data['product_qty']
+            if not self.prefill_counted_quantity:
```

Comment:
Can't we have better solution to prevent fetching `product_qty` when `prefill_counted_quantity` is false.

Reply:
We need `product_qty` to set `theoretical_qty` anyway.